src/box/iproto.cc

```diff
@@ -190,6 +194,11 @@ struct iproto_thread {
 	struct evio_service binary;
 	/** Requests count currently pending in stream queue. */
 	size_t requests_in_stream_queue;
+	/*
+	 * The maximum length of the `pending_requests` queue
+	 * in the stream, for all time.
+	 */
+	size_t historical_stream_queue_length_max;
```

Comment:
You need to reset this stat in iproto_reset_stat. Please do and add a test case for this.

Reply:
Fixed and add test case for this, but i have a question. After this change this variable is accessed from tx and from iproto thread, should we use `__ATOMIC_RELAXED` for all accesses (most correctly, but disfigures the code), or we not need atomic for it?

Reply:
No, please don't use atomics. I think we should remove atomics from rmean as well (see #6447). The statistics should be reset via a cubs call, similarly to how you transfer them from iproto to tx.

Reply:
Reset last patch.